### PR TITLE
Bug 1707573: *: remove deprecated EtcdCA and promote EtcdSigner

### PR DESCRIPTION
**Pull request by hexfusion**
Currently, etcd has 2 CAs `EtcdCA` and  EtcdSigner . This PR removed the deprecated `EtcdCA`  and promotes `EtcdSigner` as the one signer for etcd server TLS assets. To maintain backward compatibility we will honor the old `etcd-client` Secret naming for ApiServer to consume.

This is important because currently, we do not store the etcd CA key on the cluster making disaster recovery very complicated. By having the key on cluster etcd server certs can be regenerated.

/cc @deads2k @wking @abhinavdahiya

### CODE REVIEW

**Comment by richm** [comment]
buildcop here - looks like the https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1720/pull-ci-openshift-installer-master-e2e-aws/5697 test failed due to a crash in the installer:

File: pkg/asset/tls/etcd.go [hunk 1]
```diff
@@ -168,5 +99,5 @@ func (a *EtcdSignerClientCertKey) Generate(dependencies asset.Parents) error {
 
 // Name returns the human-friendly name of the asset.
 func (a *EtcdSignerClientCertKey) Name() string {
-	return "Certificate (etcd-signer-client)"
+	return "Certificate (etcd-client)"
```
Comment (abhinavdahiya) [inline pkg/asset/tls/etcd.go]: also here https://github.com/openshift/installer/pull/1720/files#diff-f19218c0ff3379e795ae4c5a125d73c9R97
:)

Comment (hexfusion) [reply pkg/asset/tls/etcd.go]: nice catch thanks

Comment (hexfusion) [reply pkg/asset/tls/etcd.go]: fixed

File: data/data/bootstrap/files/usr/local/bin/bootkube.sh.template [hunk 2]
```diff
@@ -256,9 +256,9 @@ until podman run \
 		--entrypoint etcdctl \
 		"${MACHINE_CONFIG_ETCD_IMAGE}" \
 		--dial-timeout=10m \
-		--cacert=/opt/openshift/tls/etcd-client-ca.crt \
-		--cert=/opt/openshift/tls/etcd-client.crt \
-		--key=/opt/openshift/tls/etcd-client.key \
+		--cacert=/opt/openshift/tls/etcd-ca-bundle.crt \
+		--cert=/opt/openshift/tls/etcd-signer-client.crt \
```
Comment (abhinavdahiya) [inline data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: after this change https://github.com/openshift/installer/pull/1720/files#diff-f19218c0ff3379e795ae4c5a125d73c9R97
`etcd-client.{crt,key}` are the correct file names.

Comment (hexfusion) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: ok right they are the same actually but no need to change.

Comment (hexfusion) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: well no, not now ack..

Comment (hexfusion) [reply data/data/bootstrap/files/usr/local/bin/bootkube.sh.template]: fixed

File: data/data/manifests/bootkube/etcd-serving-ca-configmap.yaml.template [hunk 3]
```diff
@@ -5,4 +5,4 @@ metadata:
   namespace: openshift-config
 data:
   ca-bundle.crt: |
-    {{.EtcdCaCert | indent 4}}
+    {{.EtcdSignerCert | indent 4}}
```
Comment (abhinavdahiya) [inline data/data/manifests/bootkube/etcd-serving-ca-configmap.yaml.template]: nit: this should be `EtcdCaBundle`

Comment (hexfusion) [reply data/data/manifests/bootkube/etcd-serving-ca-configmap.yaml.template]: done

**Comment by wking** [comment]
From the bootstrap gather:

```console
$ wget https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1720/pull-ci-openshift-installer-master-e2e-aws/5706/artifacts/e2e-aws/installer/bootstrap-logs.tar
$ tar xf bootstrap-logs.tar
$ tail -n3 control-plane/ip-10-0-137-34.ec2.internal/containers/etcd-member.log
2019-05-07 20:05:04.609226 I | embed: rejected connection from "10.0.137.34:53114" (error "remote error: tls: bad certificate", ServerName "etcd-0.ci-op-l9xy353n-1d3f3.origin-ci-int-aws.dev.rhcloud.com")
2019-05-07 20:07:11.852718 I | mvcc: store.index: compact 10800
2019-05-07 20:07:11.858636 I | mvcc: finished scheduled compaction at 10800 (took 4.682041ms)
```

So something seems broken here.

**Comment by hexfusion** [comment]
> So something seems broken here.

@abhinavdahiya found 

```
--- a/pkg/asset/manifests/operators.go
+++ b/pkg/asset/manifests/operators.go
@@ -164,7 +164,7 @@ func (m *Manifests) generateBootKubeManifests(dependencies asset.Parents) []*ass
 
        templateData := &bootkubeTemplateData{
                CVOClusterID:               clusterID.UUID,
-               EtcdCaBundle:               base64.StdEncoding.EncodeToString(etcdCABundle.Cert()),
+               EtcdCaBundle:               string(etcdCABundle.Cert()),
```

**Comment by abhinavdahiya** [comment]
/lgtm

This should be okay to merge if it goes green. :)

**Comment by hexfusion** [comment]
> fail [github.com/openshift/origin/test/extended/machines/machines.go:98]: May  7 21:09:24.210: Machine resources missing for nodes: host-10-0-0-9

/test e2e-openstack

**Comment by abhinavdahiya** [comment]
[e2e-aws](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1720/pull-ci-openshift-installer-master-e2e-aws/5709): known flake :(
```
Failing tests:

[Feature:Builds][Conformance] oc new-app  should succeed with a --name of 58 characters [Suite:openshift/conformance/parallel/minimal]
```

/retest

**Comment by abhinavdahiya** [comment]
[e2e-aws-upgrade](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1720/pull-ci-openshift-installer-master-e2e-aws-upgrade/291) failing due to load balancer not getting setup:

```
May  7 21:16:00.443: INFO: Got error testing for reachability of http://ae83e7799710c11e9bc730aa36987d0d-113645526.us-east-1.elb.amazonaws.com:80/echo?msg=hello: Get http://ae83e7799710c11e9bc730aa36987d0d-113645526.us-east-1.elb.amazonaws.com:80/echo?msg=hello: dial tcp: lookup ae83e7799710c11e9bc730aa36987d0d-113645526.us-east-1.elb.amazonaws.com on 10.142.15.200:53: no such host
May  7 21:16:02.443: INFO: Got error testing for reachability of http://ae83e7799710c11e9bc730aa36987d0d-113645526.us-east-1.elb.amazonaws.com:80/echo?msg=hello: Get http://ae83e7799710c11e9bc730aa36987d0d-113645526.us-east-1.elb.amazonaws.com:80/echo?msg=hello: dial tcp: lookup ae83e7799710c11e9bc730aa36987d0d-113645526.us-east-1.elb.amazonaws.com on 10.142.15.200:53: no such host
STEP: continuously hitting the pod through the service's LoadBalancer
```

/retest

**Pull request merged**